extensions/wikia/Chat2/js/ChatEntryPoint.js

```diff
@@ -116,54 +125,56 @@ var ChatEntryPoint = {
 	},
 
 	// change the user list into the carousel
-	initCarousel: function($el) {
+	initCarousel: function ($el) {
+		var popoverTimeout = 0,
+			// differ number of users on chat according to it's width
+			isWideChat = $el.width() > 260;
+
+
 		$el.find('.carousel-container').carousel({
 			nextClass: 'arrow-right',
 			prevClass: 'arrow-left',
-			itemsShown: (window.wgOasisResponsive) ? 5 : 6
+			itemsShown: isWideChat ? 6 : 5
 		});
 
-		// TODO: abstract this because we use this pattern in a few places: i.e. hovering over popover will not close it
-		var popoverTimeout = 0;
-
 		function setPopoverTimeout(elem) {
-			popoverTimeout = setTimeout(function() {
+			popoverTimeout = setTimeout(function () {
 				elem.popover('hide');
 			}, 300);
 		}
 
 		$el.find('.chatter').popover({
-			trigger: "manual",
-			placement: "bottom",
-			content: function() {
+			trigger: 'manual',
+			placement: 'bottom',
+			content: function () {
 				var userStatsMenu = $(this).find('.UserStatsMenu');
 
 				return userStatsMenu.clone().wrap('<div>').parent().html();
 			}
+		}).on('mouseenter', function () {
+			clearTimeout(popoverTimeout);
+			$('.popover').remove();
+			$(this).popover('show');
 
-		}).on('mouseenter', function() {
-				clearTimeout(popoverTimeout);
-				$('.popover').remove();
-				$(this).popover('show');
-
-		}).on('mouseleave', function() {
+		}).on('mouseleave', function () {
 			var $this = $(this);
+
 			setPopoverTimeout($this);
-			$('.popover').mouseenter(function() {
-				clearTimeout(popoverTimeout);
-			}).mouseleave(function() {
+			$('.popover')
+				.mouseenter(function () {
```

Comment:
because here it's chain of functions, like here in conventions: https://github.com/Wikia/guidelines/blob/master/JavaScript/CodingConventions.md#chained-method-calls